integrations/sharepoint-excel/src/index.ts

```
@@ -244,49 +233,222 @@ export default new bp.Integration({
+                await realClient.deleteTableRows({
+                  table: tableId,
+                  deleteAllRows: true,
+                })
+                logger
+                  .forBot()
+                  .info(`Successfully cleared all rows from table "${tableNameForSheet}" (ID: ${tableId}).`)
+              } catch (deleteError: any) {
+                logger
+                  .forBot()
+                  .error(
+                    `Error clearing rows from table "${tableNameForSheet}" (ID: ${tableId}): ${deleteError.message}`,
+                    deleteError.stack
+                  )
+                logger.forBot().warn(`This may result in duplicate data if old rows weren't properly cleared.`)
+              }
+
               logger
                 .forBot()
                 .info(`Populating table "${tableNameForSheet}" (ID: ${tableId}) with ${rowsData.length} new rows.`)
 
               // Get the schema to know the column types
               let tableSchema: any = null
               if (foundTable) {
-                // If we found an existing table, get its schema
+                // If we found an existing table or just created one, get its schema
                 try {
-                  const tableDetailsResponse = await axios.get(`${apiBaseUrl}/${tableId}`, { headers: httpHeaders })
-                  tableSchema = tableDetailsResponse.data.table?.schema
-                } catch (error) {
-                  logger.forBot().warn('Could not fetch table schema, will use string types for all columns')
+                  const tableDetailsResponse = await realClient.getTable({ table: tableId })
+                  tableSchema = tableDetailsResponse.table?.schema
+                } catch (error: any) {
+                  logger
+                    .forBot()
+                    .warn(
+                      `Could not fetch table schema for "${tableNameForSheet}": ${error.message}. Will use string types for all columns.`
```

Ermek - question: Will silent fail prevent user from knowing if tehre is an issue?